File: MiniDOMOffice/Presenze/Marcature/Marcature.vb
Imports minidom.Anagrafica
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Office
Imports minidom.Internals


Namespace Internals


    Public NotInheritable Class CMarcatureClass
        Inherits CModulesClass(Of MarcaturaIngressoUscita)

        Friend Sub New()
            MyBase.New("modOfficeIngressiUscite", GetType(MarcatureIngressoUscitaCursor))
        End Sub


    End Class
End Namespace

Partial Class Office

    Private Shared m_Marcature As CMarcatureClass = Nothing

    Public Shared ReadOnly Property Marcature As CMarcatureClass
        Get
            If (m_Marcature Is Nothing) Then m_Marcature = New CMarcatureClass
            Return m_Marcature
        End Get
    End Property

End Class

File: MiniDOMOffice/Presenze/Marcature/Marcature.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;
 
namespace minidom
{
    namespace repositories
    {

        /// <summary>
        /// Repository di <see cref="MarcaturaIngressoUscita"/>
        /// </summary>
        [Serializable]
        public sealed class CMarcatureClass 
            : CModulesClass<MarcaturaIngressoUscita>
        {

            /// <summary>
            /// Costruttore
            /// </summary>
            public CMarcatureClass() 
                : base("modOfficeIngressiUscite", typeof(minidom.Office.MarcatureIngressoUscitaCursor))
            {
            }
        }
    }

    public partial class Office
    {
        private static CMarcatureClass m_Marcature = null;

        /// <summary>
        /// Repository di <see cref="MarcaturaIngressoUscita"/>
        /// </summary>
        public static CMarcatureClass Marcature
        {
            get
            {
                if (m_Marcature is null)
                    m_Marcature = new CMarcatureClass();
                return m_Marcature;
            }
        }
    }
}